src/Models/NonhydrostaticModels/boundary_mass_fluxes.jl

```diff
@@ -1,26 +1,59 @@
 using Oceananigans.BoundaryConditions: BoundaryCondition, Open, PerturbationAdvection, FlatExtrapolation
-using Oceananigans.AbstractOperations: Average
-using Oceananigans.Fields: Field, interior
+using Oceananigans.AbstractOperations: Integral
+using Oceananigans.Fields: Field, interior, XFaceField, YFaceField, ZFaceField
 using Statistics: mean, filter
 using CUDA: @allowscalar
 
-@inline nanmean(arr) = mean(filter(!isnan, arr))
-@inline nanmean(num::Number) = num
-
 const OBC  = BoundaryCondition{<:Open} # OpenBoundaryCondition
 const MatchingScheme = Union{FlatExtrapolation, PerturbationAdvection}
 const ROBC = BoundaryCondition{<:Open{<:MatchingScheme}} # Radiation OpenBoundaryCondition
 const FOBC = BoundaryCondition{<:Open{<:Nothing}} # "Fixed-velocity" OpenBoundaryCondition (with no matching scheme)
 
-# Left boundary averages for normal velocity components
-@inline west_average(u)   = interior(u, 1, :, :)
-@inline south_average(v)  = interior(v, :, 1, :)
-@inline bottom_average(w) = interior(w, :, :, 1)
+# Left boundary integrals for normal velocity components
+@inline west_integral(u)   = Field(Integral(view(u, 1, :, :), dims=(2, 3)))
+@inline south_integral(v)  = Field(Integral(view(v, :, 1, :), dims=(1, 3)))
+@inline bottom_integral(w) = Field(Integral(view(w, :, :, 1), dims=(1, 2)))
+
+# Right boundary integrals for normal velocity components
+@inline east_integral(u)   = Field(Integral(view(u, u.grid.Nx + 1, :, :), dims=(2, 3)))
+@inline north_integral(v)  = Field(Integral(view(v, :, v.grid.Ny + 1, :), dims=(1, 3)))
+@inline top_integral(w)    = Field(Integral(view(w, :, :, w.grid.Nz + 1), dims=(1, 2)))
+
+function get_west_area(grid)
+    f = XFaceField(grid); set!(f, 1) # Create an XFaceField with all ones
+    ∫f = Field(Integral(f, dims=(2, 3))) # Integrate over y and z dimensions
+    @allowscalar return ∫f[1, 1, 1]
+end
```

Comment:
Isn't this replicating things that are already done within `Average`? I'm thinking of these lines:

https://github.com/CliMA/Oceananigans.jl/blob/b400322ce9602a2faadc18c4c6a085f1bcc5aa38/src/AbstractOperations/metric_field_reductions.jl#L53-L58

Reply:
Yes, but for some reason it's not very performant there. Maybe because the normalization (i.e. `dx / L`) is embedded pointwise. This creates more operations than what I do which is integrate the field once, and only then divide by L... ?

Reply:
Actually, I think it's because L is a reduced Field (I think). Maybe extracting the `Number` there is enough to speed things up. I'll apply this and report back.